### PR TITLE
feat(Snackbar): update snackbar duration logic

### PR DESCRIPTION
**Pull request by marcoskolodny**
BREAKING CHANGE: snackbar duration prop type has changed

Issue: [Link](https://jira.tid.es/browse/WEB-1996)

### CODE REVIEW

File: src/__stories__/snackbar-story.tsx [hunk 1]
```diff
@@ -9,13 +9,13 @@ export default {
...
     message: string;
-    duration: string;
+    duration: 'PERSISTENT' | 'undefined';
     type: React.ComponentProps<typeof Snackbar>['type'];
     withDismiss: boolean;
 };
 
 export const Default: StoryComponent<Args> = ({buttonText, message, duration, type, withDismiss}) => {
-    const snackbarDuration = duration !== 'Default' ? +duration : undefined;
+    const snackbarDuration = duration === 'undefined' ? undefined : duration;
```
Comment (pladaria) [inline src/__stories__/snackbar-story.tsx:18]: cosmetic change
```suggestion
    const snackbarDuration = duration === 'PERSISTENT' ? duration : undefined;
```

File: src/snackbar.tsx [hunk 2]
```diff
@@ -204,7 +204,7 @@ const Snackbar = React.forwardRef<ImperativeHandle & HTMLDivElement, Props>(
         ref
     ) => {
         const defaultDuration = buttonText ? DEFAULT_DURATION_WITH_BUTTON : DEFAULT_DURATION_WITHOUT_BUTTON;
-        duration = Math.max(duration ?? defaultDuration, defaultDuration);
+        const durationInMs = duration === 'PERSISTENT' ? Infinity : defaultDuration;
```
Comment (pladaria) [inline src/snackbar.tsx:207]: not in ms

File: src/snackbar.tsx [hunk 3]
```diff
@@ -40,7 +40,7 @@ export type ImperativeHandle = {
     close: SnackbarCloseHandler;
 };
 
-const SnackbarComponent = React.forwardRef<ImperativeHandle, Props>(
+const SnackbarComponent = React.forwardRef<ImperativeHandle, Omit<Props, 'duration'> & {duration: number}>(
```
Comment (pladaria) [inline src/snackbar.tsx:43]: why? The snackbar component should have the same duration type as the bridge implementation

File: src/__stories__/snackbar-story.tsx [hunk 4]
```diff
@@ -35,12 +35,12 @@ Default.args = {
...
     withDismiss: false,
-    duration: 'Default',
+    duration: 'undefined',
 };
 
 Default.argTypes = {
     duration: {
-        options: ['Default', 'Infinity'],
+        options: ['PERSISTENT', 'undefined'],
         control: {type: 'select'},
```
Comment (atabel) [inline src/__stories__/snackbar-story.tsx:44]: this could be a switch instead

**Review by atabel — APPROVED** [review]

File: src/snackbar.tsx (hunk 3 above)
Comment (marcoskolodny) [reply src/snackbar.tsx:43]: This is an internal component, and not used anywhere else. But I can change it just to keep the code cleaner

Comment (marcoskolodny) [reply src/snackbar.tsx:43]: updated

File: src/__stories__/snackbar-story.tsx (hunk 4 above)
Comment (marcoskolodny) [reply src/__stories__/snackbar-story.tsx]: updated

File: src/snackbar.tsx (hunk 2 above)
Comment (marcoskolodny) [reply src/snackbar.tsx]: done

**Review by pladaria — APPROVED** [review]

**Pull request merged**